packages/x-codemod/src/util/removeObjectProperty.ts

```diff
@@ -0,0 +1,62 @@
+import type { Collection, JSCodeshift } from 'jscodeshift';
+
+interface RemoveObjectPropertyArgs {
+  root: Collection<any>;
+  componentsNames: string[];
+  /**
+   * Prop which contains the object whose property will be removed
+   * @example "experimentalFeatures"
+   */
+  propName: string;
+  /**
+   * `key` of the property that needs to be removed
+   * To remove `newEditingApi` from:
+   * <DataGrid experimentalFeatures={{ newEditingApi: true }} />
+   * pass "newEditingApi"
+   */
+  propertyToRemove: string;
```

Comment:
Nitpick: `propKey` would probably be enough given the function (name) context. 🤔